assigner/commands/get.py

```diff
@@ -23,23 +27,76 @@ def get(conf, args):
     namespace = conf.namespace
     token = conf.gitlab_token
     semester = conf.semester
+    branch = args.branch
+    force = args.force
 
     roster = get_filtered_roster(conf.roster, args.section, args.student)
 
     path = os.path.join(hw_path, hw_name)
     os.makedirs(path, mode=0o700, exist_ok=True)
 
-    count = 0
-    for student in roster:
+    output = PrettyTable(["#", "Sec", "SID", "Name", "Change"], print_empty=False)
+    output.align["Name"] = "l"
+    output.align["Change"] = "l"
+
+    progress = tqdm()
+
+    for i,student in progress.enumerate(roster):
         username = student["username"]
         student_section = student["section"]
         full_name = StudentRepo.name(semester, student_section,
-                                     hw_name, username)
+                                    hw_name, username)
 
         try:
             repo = StudentRepo(host, namespace, full_name, token)
-            repo.clone_to(os.path.join(path, username))
-            count += 1
+            repo_dir = os.path.join(path, username)
+
+            row = str(i + 1)
+            sec = student["section"]
+            sid = student["username"]
+            name = student["name"]
+
+            try:
+                logging.debug("Attempting to use local repo {}...".format(repo_dir))
+                repo.add_local_copy(repo_dir)
+
+                logging.debug("Local repo exists, fetching...")
+                results = repo.repo.remote().fetch()
+                for result in results:
+                    logging.debug("fetch result: {} {} {}".format(result.ref.name, result.flags, result.note))
+
+                    if result.flags & result.NEW_HEAD:
```

Comment:
🤢 
I'm surprised and upset that this is the way to do this?

Reply:
That's what you get for using a library that doesn't wrap git tightly enough!

Reply:
I will document this though.